disk/disk.h

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef PV_DISK_H
+#define PV_DISK_H
+
+#include "utils/list.h"
+#include "state.h"
+
+typedef enum {
+	DISK_UNKNOWN,
+	DISK_DIR,
+	DISK_DM_CRYPT_VERSATILE,
+	DISK_DM_CRYPT_CAAM,
+	DISK_DM_CRYPT_DCP,
+	DISK_SWAP,
+	DISK_VOLUME
+} pv_disk_t;
+
+typedef enum {
+	DISK_FORMAT_UNKNOWN,
+	DISK_FORMAT_SWAP,
+	DISK_FORMAT_EXT4,
+	DISK_FORMAT_EXT3
+} pv_disk_format_t;
+
+typedef enum {
+	DISK_STATUS_ERROR,
+	DISK_STATUS_MOUNTED,
+	DISK_STATUS_NOT_MOUNTED
+} pv_disk_status_t;
+
+struct pv_disk {
+	char *name;
+	pv_disk_t type;
+	pv_disk_format_t format;
+	char *path;
+	char *uuid;
+	char *mount_ops;
+	char *format_ops;
+	char *provision_ops;
+	char *provision;
+	char *mount_target;
+	bool def;
+	bool mounted;
+	// pv_disk
+	struct dl_list list;
+};
+
+struct pv_disk *pv_disk_add(struct pv_state *s);
```

Comment:
I might be wrong in this one but I think we should switch all functions in disk.h interface from working with pv_state into work with pv_disk in the same way we do with pv_volume in volumes.c, and leave all state level stuff to state.c . This works better in the context of the comment that I left in state.c and the spirit of working with disks individually instead of state-wide.

Reply:
Yeah, that will be nice. Let me check if it's possible. I thought on this too, but at one point I forgot

Reply:
Check https://github.com/pantavisor/pantavisor/pull/363